### PR TITLE
Change all exercises to use Swift package manager

### PR DESCRIPTION
**Pull request by lyuha**
Related #233 #142

### CODE REVIEW

File: exercises/accumulate/Tests/accumulateTests/accumulateTests.swift [hunk 1]
```diff
@@ -21,10 +19,8 @@ private extension XCTest {
     }
 }
 
-class AccumulateTest: XCTestCase {
-
+class accumulateTests: XCTestCase {
```
Comment (masters3d) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: classes, or Types need to remain capitalized

File: exercises/accumulate/Tests/accumulateTests/accumulateTests.swift [hunk 2]
```diff
@@ -95,4 +82,14 @@ class AccumulateTest: XCTestCase {
 
         XCTAssertEqualMultiArray(expected, result)
     }
+
+    static var allTests: [(String, (accumulateTests) -> () throws -> Void)] {
```
Comment (masters3d) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: nice! this is great. we can run on linux now

File: exercises/allergies/Tests/allergiesTests/allergiesTests.swift [hunk 3]
```diff
@@ -1,8 +1,7 @@
-#if swift(>=3.0)
-    import XCTest
-#endif
+import XCTest
+@testable import allergies
 
-class AllergiesTest: XCTestCase {
+class allergiesTests: XCTestCase {
```
Comment (masters3d) [inline exercises/allergies/Tests/allergiesTests/allergiesTests.swift]: capitalize

**Comment by masters3d** [comment]
Could you update the xcode project so that travis can run the tests? 

Could we make one package that contains all packages so we can run tests on linux?


This is great!

File: exercises/accumulate/Tests/accumulateTests/accumulateTests.swift (hunk 2 above)
Comment (lyuha) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: But, tournament can't pass test only on Linux.

File: exercises/accumulate/Tests/accumulateTests/accumulateTests.swift (hunk 1 above)
Comment (lyuha) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: Sorry, It is made by swift package manager, I change all exercises test.

Comment (lyuha) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: Could I change filename and package name with CamelCase style?

File: exercises/accumulate/Tests/accumulateTests/accumulateTests.swift (hunk 2 above)
Comment (masters3d) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: We can fix the non passing ones on another PR.

File: exercises/accumulate/Tests/accumulateTests/accumulateTests.swift (hunk 1 above)
Comment (masters3d) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: You can change the test packages names.

Comment (masters3d) [inline exercises/accumulate/Tests/accumulateTests/accumulateTests.swift]: Only the main folder per excersice needs to keep its name so the script can find and serve on command line.

**Comment by masters3d** [comment]
we could also eliminate the xcode project and just use a genratec xcode project to run the test. i am going to create a new branch to merge this PR so we can test this before merging into master.

**Comment by lyuha** [comment]
I tested all exercises with `swift test` on my macOS.
Only testIgnoreBadlines in tournament is failed.

**Comment by masters3d** [comment]
Let me know when it is ready and we can merge this part and I'll fix travis later.

File: exercises/accumulate/Tests/AccumulateTests/AccumulateTests.swift [hunk 4]
```diff
@@ -95,4 +82,14 @@ class AccumulateTest: XCTestCase {
...
         XCTAssertEqualMultiArray(expected, result)
     }
+
+    static var allTests: [(String, (AccumulateTests) -> () throws -> Void)] {
+        return [
+            ("testEmptyAccumulation", testEmptyAccumulation),
+            ("testAccumulateSquares", testAccumulateSquares),
+            ("testAccumulateUpcases", testAccumulateUpcases),
+            ("testAccumulateReversedStrings", testAccumulateReversedStrings),
+            ("testAccumulateRecursively", testAccumulateRecursively),
```
Comment (masters3d) [inline exercises/accumulate/Tests/AccumulateTests/AccumulateTests.swift]: dont worry about these. Ill modify the rule

**Comment by lyuha** [comment]
Thanks

**Comment by masters3d** [comment]
Thank you!

File: exercises/grade-school/Tests/GradeSchoolTests/GradeSchoolTests.swift [hunk 5]
```diff
@@ -20,12 +19,6 @@ private extension XCTestCase {
         XCTAssert(sameCollection(collectionS, collectionA))
     }
 
-    func XCTAssertEqualCollection (_ collectionA1: [String]?, _ collectionA2: [String]? ) {
```
Comment (masters3d) [inline exercises/grade-school/Tests/GradeSchoolTests/GradeSchoolTests.swift]: This needs to go back in to accommodate array to array comparisons https://github.com/exercism/xswift/issues/258

**Pull request merged**